### PR TITLE
Added support for Billing app navigation to approved Admin settings

### PR DESCRIPTION
**Pull request by aileen**
closes [GVA-868](https://linear.app/ghost/issue/GVA-868/allow-billing-to-navigate-to-approved-ghost-admin-settings)

## What

Ghost Admin now accepts a semantic navigation request from the validated Billing iframe:

```
{request: 'navigateToAdmin', destination: '<approved destination>'}
```

Admin owns the destination→route mapping (allowlist only — no raw URLs, no reuse of the existing `route` sync field):

| Destination | Route |
|---|---|
| `theme` | `/settings/design/change-theme` (theme chooser) |
| `analytics` | `/settings/analytics` |
| `staff` | `/settings/staff` |
| `stripe` | `/settings/stripe-connect` |
| `integrations` | `/settings/integrations` |
| `newsletters` | `/settings/emails` with the `automations` labs flag, otherwise `/settings/newsletters` |

## How

- The request is handled in `gh-billing-iframe` only after the existing `isValidBillingIframeMessage` origin + source-window validation, alongside the other request handlers.
- `BillingService.navigateToAdminDestination()` resolves the destination against a frozen, null-prototype allowlist (so keys like `__proto__`/`constructor` cannot resolve) and navigates via `router.transitionTo()`. Unknown, missing, or non-string destinations are ignored silently.
- Leaving `/pro` through client-side routing closes the Billing overlay via the existing `pro` route `willTransition` handling and preserves usable browser history.
- Registered the `automations` labs flag in the Ember `feature` service (it previously only existed in the React admin).

## Testing

- 12 new integration tests at the postMessage boundary (all six destinations incl. both newsletters variants, unknown/missing/malformed destinations, invalid origin, wrong source window).
- 3 new unit tests for the service method incl. prototype-pollution keys.
- Existing billing route-sync tests unchanged and green; `feature` service suite green; lint clean.
- Mutation-checked: changing a route string or inverting the automations condition makes the suite fail.

## Release note

⚠️ This must be shipped and released **before** the Billing App starts emitting `navigateToAdmin` requests (unblocks GVA-869).

### CODE REVIEW

**Comment by aileen** [comment]
Added a second commit: **🐛 Fixed stale billing limit data when reopening the billing screen**

`limitUpdate` was only sent from the nav-link entry path (`openBillingWindow`), so returning to `#/pro` via browser history or direct URL rendered stale limit data in the Billing app — which becomes routine once requirement-action round-trips (GVA-869) ship. The send now lives in `toggleProWindow` on the actual hidden→visible transition (single source of truth for every entry path); re-entrant opens and overlay hides send nothing, and the nav-link path sends exactly once.

**Verification**
- 6 new unit tests on the billing service (show sends once / hide none / unloaded iframe none / redundant re-open none / checkout re-entry none / nav-link exactly once); mutation-checked on both axes.
- Manually verified against a live env (Ghost dev + BMA on :8888, staging Daisy): instrumented `sendUpdateLimits` in the running admin — history-back into `#/pro` after a `navigateToAdmin` round-trip fired exactly one send (previously zero), nav-link entry fired exactly one, leaving the overlay fired none. Also confirmed the GVA-868 bridge end-to-end: the Billing app's requirement link navigated Admin to `#/settings/analytics`, closed the overlay, and history stayed usable.

File: ghost/admin/tests/unit/services/billing-test.js [hunk 1]
```diff
@@ -342,4 +342,120 @@ describe('Unit: Service: billing', function () {
...
+        expect(limitUpdateCalls(postMessage)).to.have.lengthOf(0);
+    });
+
+    it('does not send limitUpdate when the billing iframe is not loaded', function () {
+        const service = this.owner.lookup('service:billing');
+        billingService = service;
+        sinon.stub(service, 'getBillingIframe').returns(null);
+
+        expect(() => service.toggleProWindow(true)).to.not.throw();
+    });
```
[the remark this answers is not in the document]
Comment (aileen) [reply ghost/admin/tests/unit/services/billing-test.js]: Strengthened in 6fed0477e26. A postMessage spy can't observe the unloaded case directly — an unloaded iframe has no `contentWindow`, so there is no window object to receive (or spy) a message, and any non-optional-chained send path throws, which the retained no-throw assertion catches. To cover the queued/deferred-send regression class, the test now restores a spy-equipped iframe after the transition, settles the runloop, and asserts zero `limitUpdate` posts. Verified: suite passes 24/24, and the assertion fails if the transition schedules a late send.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
